pkgs/by-name/de/dewduct/package.nix

```diff
@@ -0,0 +1,94 @@
+{ lib
+, cargo
+, fetchFromGitHub
+, glib
+, gtk4
+, libadwaita
+, makeDesktopItem
+, mpv
+, openssl
+, pkg-config
+, rustc
+, rustPlatform
+, stdenv
+, wrapGAppsHook4
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "dewduct";
+  version = "v0.2.2";
+
+  src = fetchFromGitHub {
+    owner = "DaKnig";
+    repo = "DewDuct";
+    rev = finalAttrs.version;
+    hash = "sha256-PezCs4UxlapXqDOuVJ5QNjMQtQle/DE2eBnVpMYMu54=";
+  };
+
+  cargoDeps = rustPlatform.fetchCargoTarball {
+    src = finalAttrs.src;
+    name = "${finalAttrs.pname}-${finalAttrs.version}";
+    hash = "sha256-dT+A96A4kTAVcEyxoZZnIKGb0NLu88IXjyTh0unxre4=";
+  };
+
+  nativeBuildInputs = [
+    cargo
+    pkg-config
+    rustPlatform.cargoSetupHook
+    rustc
+    wrapGAppsHook4
+  ];
+
+  buildInputs = [
+    glib
+    gtk4
+    libadwaita
+    mpv
+    openssl
+  ];
+
+  preFixup = ''
+    gappsWrapperArgs+=(
+      --prefix PATH : "${mpv}/bin"
+    )
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/bin
+    cp target/release/dewduct $out/bin/
+
+    mkdir -p $out/share
+    ln -s ${finalAttrs.desktopItem}/share/applications $out/share/
+
+    runHook postInstall
+  '';
+
+  desktopItem = makeDesktopItem {
+    name = finalAttrs.pname;
+    exec = finalAttrs.pname;
```

Comment:
Referencing pname everywhere is generally undesirable, see #277994

Reply:
Thanks for this :slightly_smiling_face: